### ISSUE TITLE
Add an optional repeatable key to the repeat options

### ISSUE THREAD

**Issue by manast** [taskforcesh/bullmq#1822]
It would be pretty useful to be able to assign a custom repeatable key (i.e. the key where the repeatable metadata is stored), instead of relying on the automatic generation of the key based on the repeat options.

**Describe the solution you'd like**
Add an optional "key" field to the RepeatOptions interface. If this option is provided, instead of generating the key based on the repeat options it should use the key as the base for the actual key stored in Redis.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by tybro0103** [comment]
It seems like this would go a long way in making it easier to update the schedule of an existing repeatable without unintentionally creating a new repeatable... or am I missing something? I landed here by trying to figure out this particular problem.

**Comment by tybro0103** [comment]
So for my use I want to be able to define a list of things in a single place that need to run on a regular schedule. Something like:
```js
const jobScheduleMap = {
  'refresh-the-cache': '0 3 * * *',
  'check-for-expired-things': '0 4 * * *',
  'process-subscriptions': '0 5 * * *',
};
```
And then I need to be able to ensure the repeatable jobs in Bull are in sync with this. So for now I'm resorting to this:
```js
// removing an item from this map will NOT cause it to be removed from the queue
// set value to 'DELETE' to cause it to be removed from the queue
const jobScheduleMap = {
  'refresh-the-cache': '0 3 * * *',
  'foo': 'DELETE',
  'check-for-expired-things': '0 4 * * *',
  'process-subscriptions': '0 5 * * *',
};

// reconciles the actual repeatables with the desired repeatables according to jobScheduleMap
const syncRepeatables = async () => {
  // STEP 1: loop through all the existing repeatable jobs
  //   remove those that are set for deletion or where the schedule isn't correct
  //   ignore those that aren't in the jobScheduleMap
  //   compile a list of those that are already correct; used in the next step
  const repeatables = await queue.getRepeatableJobs();
  const alreadyCreatedList = [];
  await Promise.all(repeatables.map(async (repeatable) => {
    const expectedPattern = jobScheduleMap[repeatable.name];
    if ((expectedPattern === 'DELETE') || (expectedPattern && (expectedPattern !== repeatable.pattern))) {
      await queue.removeRepeatableByKey(repeatable.key);
    } else if (expectedPattern === repeatable.pattern) {
      alreadyCreatedList.push(repeatable.name);
    }
  }))
  // STEP 2: loop through the items in jobScheduleMap
  //   create repeatables if not already created and not marked for deletion
  await Promise.all(Object.entries(jobScheduleMap).map(async ([jobName, schedule]) => {
    const alreadyCreated = alreadyCreatedList.includes(jobName);
    const markedForDeletion = schedule === 'DELETE';
    if (!alreadyCreated && !markedForDeletion) {
      await queue.add(jobName, null, {repeat: {pattern: schedule}});
    }
  }));
};
```
And executing that during deploy time.

Clearly there's a bunch of issues (like if I want to specify other options or add job data). It could be tweaked to take more things like that into account, but... It seems like being able to add a repeatable to the queue with a specified key and Bull figuring out how to update it would be preferred.

**Comment by luketych** [comment]
Please please please make this happen 💯

**Comment by EinfachHans** [comment]
I see there is already a PR for this, any plans to merge it?

**Comment by Marsup** [comment]
Hi,

I see that this issue was closed with #1824. I was under the impression the goal was to be able to have calls such as `removeRepeatableByKey(key)` with the same key, like mentioned above by @tybro0103. It doesn't seem to work this way, but reading the code of the PR maybe it was never the intention, as it's not part of the unit tests. Am I missing something, or is there still no way to delete a repeatable job with a static key?

**Comment by EinfachHans** [comment]
@Marsup seems to be right, doesn't working that way...

Was that ever the intention? It would be awesome to find, add and delete delayed & repeatable jobs via the same key
